modules/setting/setting.go

```diff
@@ -774,6 +775,24 @@ func NewContext() {
 
 	InternalToken = loadInternalToken(sec)
 
+	var dictPC = map[string]string{
+		"lower": "[a-z]+",
+		"upper": "[A-Z]+",
+		"digit": "[0-9]+",
+		"spec":  "[-_]+",
+	}
+	PasswordComplexity = make(map[string]string)
+	cfgdata := sec.Key("PASSWORD_COMPLEXITY").Strings(",")
```

Comment:
but If cfgdata is empty, PasswordComplexity wasn't given the default value? So as my above comment, your default value of `PASSWORD_COMPLEXITY` is empty that means don't check password complexity but not `lower,upper,digit,spec`.